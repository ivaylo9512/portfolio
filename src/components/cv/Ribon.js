const Ribon = ({ name, color }) => {
    return(
        <div className={color}>
            {name}
        </div>
    )
}

export default Ribon