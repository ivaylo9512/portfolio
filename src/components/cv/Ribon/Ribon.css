.ribon{
    height: 3.3vw;
    background: black;
    border-radius: 0.3vw;
}