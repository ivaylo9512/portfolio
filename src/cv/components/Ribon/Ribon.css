.ribon{
    display: block;
    width: 98%;
    margin-top: 3.2%;
    position: relative;
}
.ribon > svg{
    display: block;
}
.ribon rect{
    filter: url(#shape-shadow);
}
.ribon .content{
    position: absolute;
    display: flex;
    width: 100%;
    height: 100%;
    padding-left: 3%;
    z-index: 5;
    box-sizing: border-box;
    align-items: center;
}
.ribon span{
    flex: 1;
}