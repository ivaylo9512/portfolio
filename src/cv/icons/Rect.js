const Rect = ({ height, width }) => {
    return(
        <svg className="rect" xmlns="http://www.w3.org/2000/svg" viewBox={`0 0 ${width} ${height}`} xmlnsXlink="http://www.w3.org/1999/xlink" >
            <rect width="100%" height="100%" rx="10" ry="10"/>
        </svg>
    )
}

export default Rect