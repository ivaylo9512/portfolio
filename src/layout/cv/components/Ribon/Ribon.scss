.ribon{
    position: relative;
    opacity: 0.9;

    > svg{
        display: block;
    }

    rect{
        filter: url(#shape-shadow);
    }

    .content{
        position: absolute;
        display: flex;
        width: 100%;
        height: 100%;
        padding-left: 3%;
        z-index: 5;
        box-sizing: border-box;
        align-items: center;
    }

    @media only screen and (min-width: 700px) {
        opacity: 1; 
    }
}