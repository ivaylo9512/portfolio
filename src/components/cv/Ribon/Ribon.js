import './Ribon.css';

const Ribon = ({ name, color, children }) => {
    return(
        <div className={'ribon ' + color}>
            <span>{name}</span>
        </div>
    )
}

export default Ribon