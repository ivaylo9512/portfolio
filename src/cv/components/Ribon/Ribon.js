import Rect from 'cv/icons/Rect';
import './Ribon.scss';

const Ribon = ({ name, children }) => {
    return(
        <div className='ribon'>
            <div className='content'>
                {name}
                {children}
            </div>
            <Rect height='85' width={564}/>
        </div>
    )
}

export default Ribon