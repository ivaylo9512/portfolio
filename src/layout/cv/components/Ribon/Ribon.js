import Rect from '@svgIcons/Rect';
import './Ribon.scss';

const Ribon = ({ name, ribon, children }) => {
    return(
        <div className='ribon'>
            <div className='content'>
                {ribon ? ribon : name ? <span>{name}</span> : ""}
                {children}
            </div>
            <Rect height='85' width={564}/>
        </div>
    )
}

export default Ribon