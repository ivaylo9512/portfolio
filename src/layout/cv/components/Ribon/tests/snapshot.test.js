import Ribon from '../Ribon';
import { shallow } from 'enzyme';
import Rect from '@svgIcons/Rect';

describe('Ribon snapshot tests', () => {
    const createWrapper = () => shallow(
        <Ribon name='test-name'>
            <Rect />
        </Ribon>
    )

    it('should match snapshot', () => {
        const wrapper = createWrapper();
     
        expect(wrapper).toMatchSnapshot();
    })
});