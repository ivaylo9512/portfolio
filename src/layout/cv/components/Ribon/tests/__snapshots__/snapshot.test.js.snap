// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribon snapshot tests should match snapshot 1`] = `
<div
  className="ribon"
>
  <div
    className="content"
  >
    test-name
    <Rect />
  </div>
  <Rect
    height="85"
    width={564}
  />
</div>
`;
